apps/testrunner/queue.py

```diff
@@ -0,0 +1,105 @@
+import json
+import time
+
+import requests
+import sys
+
+from .logger import logger
+from .runner import SimpleTest
+
+
+class TaskRepo(object):
+    # todo: add HTTP requests to fetch/save data
+    URL = 'http://localhost:8000/task'
+
+    def acquire(self):
+        url = '{}/acquire'.format(self.URL)
+        response = requests.get(url)
+        if response.ok:
+            task_data = response.json()
+            return Task(self, task_data['id'], task_data)
+
+    def release(self, task):
+        url = '{}/{}/lock'.format(self.URL, task.id)
+        response = requests.delete(url)
+        response.raise_for_status()
+
+    def submit_result(self, task, result):
+        url = '{}/{}/result'.format(self.URL, task.id)
+        response = requests.post(url, json=json.dumps(result))
+        response.raise_for_status()
+
+
+class Task(dict):
+    def __init__(self, repo, id, *args, **kwargs):
+        self.repo = repo
```

Comment:
It will not overwrite them since object "attributes" and "items" are two different things, eg.
```
In [5]: a = C({'id':1})
In [6]: a.id = 2
In [7]: a
Out[7]: {'id': 1}
In [8]: a.id
Out[8]: 2
```

however this still would me misleading though, I'm going to create properties for "repo" and "id"